packages/perseus-core/src/widgets/matcher/matcher-util.ts

```diff
@@ -0,0 +1,31 @@
+import type {PerseusMatcherWidgetOptions} from "@khanacademy/perseus-core";
+
+/**
+ * For details on the individual options, see the
+ * PerseusMatcherWidgetOptions type
+ */
+type MatcherPublicWidgetOptions = {
+    labels: PerseusMatcherWidgetOptions["labels"];
+    left: Array<string>;
+    right: Array<string>;
```

Comment:
Will this be a problem to still work with the original widget options? @jeremywiebe, I remember you talking about how it should be backward compatible in a way so that the public widget options can still pass where the widget options are expected, or something like that.

Reply:
If you use `ReadonlyArray` instead of `Array` here, do you get a type error? If you can use `ReadonlyArray`, I agree that we might want to prefer that for compatibility with the other types.

Reply:
Where possible, I like using `ReadonlyArray<>` because these types represent data that Perseus will receive. The idea that it could be mutable and mutated should be private to the widget. So yes, I'd agree with Ben to try switching these to `ReadonlyArray<>`.  If that causes type issues, then I think we should look to fix the function code and not the type.

Reply:
Switched back to reference the original widget options type :)